source/common/http/conn_manager_impl.cc

```diff
@@ -322,15 +323,26 @@ Network::FilterStatus ConnectionManagerImpl::onData(Buffer::Instance& data, bool
   bool redispatch;
   do {
     redispatch = false;
+    Status status;
 
     try {
-      codec_->dispatch(data);
+      status = codec_->dispatch(data);
+      // TODO(#10878): Remove this when exception removal is complete. It is currently in migration.
+      // Soon we won't need to catch these exceptions, as they'll be propagated through the error
+      // statuses callbacks and returned from dispatch.
     } catch (const FrameFloodException& e) {
-      handleCodecException(e.what());
-      return Network::FilterStatus::StopIteration;
+      status = bufferFloodError(e.what());
     } catch (const CodecProtocolException& e) {
+      status = codecProtocolError(e.what());
+    }
```

Comment:
can we factor out this stanza from codec_client.c? It looks identical.

The stanza below is different of course but this looks like it would make a good helper function callable in both places.

Reply:
qq before I make this change (as it's pretty big) about where to factor this out to:
1. utility function that takes codec and data. pros: will be simple to remove when exception removal, but it is kind of clunky.
2. I could factor this out so ConnectionImpl::dispatch implemented in codec_impl's is the actual try/catch block around an `innerDispatch` that currently does what dispatch does. This dispatch /always/ returns a status correctly and no one has to worry about whether it throws or not. Also has the advantage of minimizing the diff between the new and legacy codec_impl_tests. calls to dispatch will always return a status, not a status sometimes and a throw for others. I am leaning to this option, but it's a mass change.

Reply:
Yeah +1 for (2) that sounds great.

Reply:
done, I wanted to at least try that suggestion out. A uniform return value from dispatch is clean and causes less confusion.

There was one big pain point, see the diff in http2/codec_impl_test.cc. This test https://github.com/envoyproxy/envoy/blob/c86c679dabe6813d779f32e0225abcd83a56b32f/test/common/http/http2/codec_impl_test.cc#L414 will fail *unless* I directly call the `innerDispatch` in the mock connection in the test fixture. Why? Because `encodeHeaders` throws, and if I directly call dispatch, I swallow the throw because `write(_,_)` does not return a status. The stack trace for the failure when that happens goes through `encodeHeaders`, `encodeHeadersBase`, `sendPendingFrames()`, and then throws. This throw *does not* occur in a dispatching context! see https://github.com/envoyproxy/envoy/blob/c86c679dabe6813d779f32e0225abcd83a56b32f/source/common/http/http2/codec_impl.cc#L893

Hence, this solution is kinda ugly because in H/2 test I had to call in inner dispatch, because we test a non-dispatching throw.